app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/mini_game/CoffeeBoostActivity.java

```diff
@@ -0,0 +1,179 @@
+package nl.tudelft.pixelperfect.pixelperfect.mini_game;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.GridLayout;
+
+import nl.tudelft.pixelperfect.client.GameClient;
+import nl.tudelft.pixelperfect.client.message.EventCompletedMessage;
+import nl.tudelft.pixelperfect.event.type.EventTypes;
+import nl.tudelft.pixelperfect.pixelperfect.R;
+import nl.tudelft.pixelperfect.pixelperfect.Spaceship;
+
+/**
+ * Mini-game for the Janitor.
+ *
+ * @author Wouter Zirkzee
+ *
+ */
+public class CoffeeBoostActivity extends Activity {
```

Comment:
This looks great. Maybe later you could create a separate component for the logic of the puzzle, instead of putting it all in the Activity?